configure.ac

```diff
@@ -2829,7 +2840,7 @@ AC_CHECK_HEADERS(
 
 # checks for typedefs
 AC_CACHE_CHECK([for clock_t in time.h], [ac_cv_clock_t_time_h], [
-  AC_EGREP_HEADER([clock_t], [time.h], [ac_cv_clock_t_time_h=yes], [ac_cv_clock_t_time_h=no])
+  AX_CHECK_TYPEDEF([clock_t], [time.h], [ac_cv_clock_t_time_h=yes], [ac_cv_clock_t_time_h=no])
```

Comment:
I'm still pretty new to autoconf, but it looks to me like the [definition of AX_CHECK_TYPEDEF](https://git.savannah.gnu.org/gitweb/?p=autoconf-archive.git;a=blob_plain;f=m4/ax_check_typedef.m4) ignores its third and fourth arguments and replaces them with AC_MSG_RESULT.

[This project](https://github.com/NumCosmo/NumCosmo/commit/203ea7469bc45ffa19e693c8149edbc5603cba22) may have hit the same problem, and responded by replacing AX_CHECK_TYPEDEF with AC_CHECK_TYPES.

Reply:
Hm, I did not study the generated output good enough. The cv variable now has a slightly different format. It seems to me should apply this patch (diff against PR):

```diff
diff --git a/configure.ac b/configure.ac
index 6b551b699d..c18343a11b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2838,11 +2838,11 @@ AC_CHECK_HEADERS(
 ])
 
 # checks for typedefs
-AC_CACHE_CHECK([for clock_t in time.h], [ac_cv_clock_t_time_h], [
-  AX_CHECK_TYPEDEF([clock_t], [time.h], [ac_cv_clock_t_time_h=yes], [ac_cv_clock_t_time_h=no])
-])
+AC_CACHE_CHECK([for clock_t in time.h],
+               [ac_cv_type_clock_t_time_h],
+               [AX_CHECK_TYPEDEF([clock_t], [time.h])])
 dnl checks for "no"
-AS_VAR_IF([ac_cv_clock_t_time_h], [no], [
+AS_VAR_IF([ac_cv_type_clock_t_time_h], [no], [
   AC_DEFINE([clock_t], [long],
             [Define to 'long' if <time.h> doesn't define.])
 ])

```

Reply:
> [This project](https://github.com/NumCosmo/NumCosmo/commit/203ea7469bc45ffa19e693c8149edbc5603cba22) may have hit the same problem, and responded by replacing AX_CHECK_TYPEDEF with AC_CHECK_TYPES.

That's also an option, but `AC_CHECK_TYPES` introduces a new definition to pyconfig.h. OTOH, `AX_CHECK_TYPEDEF` introduces a new macro to aclocal.m4.

Reply:
Another nice thing with `AC_CHECK_TYPES`, is that it has caching automatically built in.

Reply:
Diff for `AC_CHECK_TYPES`:

```diff
diff --git a/configure.ac b/configure.ac
index 6b551b699d..3e1ea3ff05 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2837,15 +2837,13 @@ AC_CHECK_HEADERS(
 #endif
 ])
 
-# checks for typedefs
-AC_CACHE_CHECK([for clock_t in time.h], [ac_cv_clock_t_time_h], [
-  AX_CHECK_TYPEDEF([clock_t], [time.h], [ac_cv_clock_t_time_h=yes], [ac_cv_clock_t_time_h=no])
-])
-dnl checks for "no"
-AS_VAR_IF([ac_cv_clock_t_time_h], [no], [
-  AC_DEFINE([clock_t], [long],
-            [Define to 'long' if <time.h> doesn't define.])
-])
+# Check for clock_t in time.h.
+AC_CHECK_TYPES([clock_t],
+               [],
+               [AS_VAR_IF([ac_cv_type_clock_t], [no],
+                          [AC_DEFINE([clock_t], [long],
+                                     ["Define to 'long' if <time.h> doesn't define."])])],
+               [@%:@include time.h])
 

```

Reply:
Is the `AS_VAR_IF` necessary? It's already in the "not found" branch.

Reply:
Hah, true 😆 I guess I was in a hurry earlier today; looks like my last commit broke CI quite bad.